src/rdb.c

```diff
@@ -3645,6 +3645,21 @@ void bgsaveCommand(client *c) {
     if (c->argc > 1) {
         if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr, "schedule")) {
             schedule = 1;
+        } else if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr, "cancel")) {
+            /* Terminates an in progress BGSAVE */
+            if (server.child_type == CHILD_TYPE_RDB) {
+                /* There is an ongoing save */
+                serverLog(LL_NOTICE, "Background bgsave will be aborted due to user request");
+                killRDBChild();
+                addReplyStatus(c, "Background bgsave cancelled");
```

Comment:
I noticed the other replies use the term "background saving" (like "Background saving scheduled").

```suggestion
                addReplyStatus(c, "Background saving cancelled");
```